### PR TITLE
Move viewer theme state management to context

### PR DESCRIPTION
**Pull request by msnyder-msft**
### Summary of changes 🔍 
Pulling out the state management for the viewer theme into a context with a reducer so that it is more portable.

No visual changes.
image

### Testing 🧪
- [x] Play with the theme picker and make sure it does what you expect

### Checklist ✔️
- [x] Linked associated issue (if present)
- [x] Added relevant labels
- [x] Requested developer reviews
- [x] Request UI review from design / PM
- [x] Verify all code checks are passing

### CODE REVIEW

File: src/Components/ModelViewerModePicker/ModelViewerModePicker.tsx [hunk 1]
```diff
@@ -134,105 +89,41 @@ const ModelViewerModePicker: React.FC<ModelViewerModePickerProps> = ({
...
-            background: backgroundColors[0].color,
-            style: ViewerModeStyles.Default
-        });
-
-        viewerModeUpdated({
-            objectColor: null,
-            background: backgroundColors[0].color,
-            style: ViewerModeStyles.Default
-        });
+        // // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (ccrowley96) [inline src/Components/ModelViewerModePicker/ModelViewerModePicker.tsx:92]: nit double comment here -- may also be able to remove this line entirely

File: src/Models/Context/SceneThemeContext/SceneThemeContext.tsx [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+
+    // skip wrapping if the context already exists
+    const existingContext = useSceneThemeContext();
+    if (existingContext) {
+        return <>{children}</>;
+    }
+
+    const { initialState, usePersistedTheme = true } = props;
+    const { t } = useTranslation();
+    shouldPersistTheme = usePersistedTheme; // store the selection globally
```
Comment (ccrowley96) [inline src/Models/Context/SceneThemeContext/SceneThemeContext.tsx:109]: nit: starting a variable name with `use` is unofficially reserved for react hooks.  Maybe rename to `isThemePersisted`?

**Review by ccrowley96 — APPROVED** [review]
Great refactor.  Code looks good.  Just left a couple nits.

File: src/Models/Context/SceneThemeContext/SceneThemeContext.tsx (hunk 2 above)
Comment (msnyder-msft) [reply src/Models/Context/SceneThemeContext/SceneThemeContext.tsx:109]: Fair. Updated.

**Review by samjosephgates — APPROVED** [review]
Looks good. The only difference in behavior that I can see is as follows:

Current implementaton: 
Select transparent style (it defaults to blue)
Change object color to gold
Change back to default style
Change back to transparent
It is still gold

New implementation
Select transparent style (it defaults to blue)
Change object color to gold
Change back to default style
Change back to transparent
It defaults back to blue

Not sure which is right, they are just different.

**Pull request merged**